server/services/lists/caCaseloadService.ts

```diff
@@ -219,287 +278,132 @@ export default class CaCaseloadService {
+      .filter(offender => !offender.prisoner.indeterminateSentence)
+      .filter(offender => offender.prisoner.conditionalReleaseDate)
       .filter(offender =>
         CaseListUtils.isEligibleEDS(
-          offender.nomisRecord.paroleEligibilityDate,
-          offender.nomisRecord.conditionalReleaseDate,
-          offender.nomisRecord.confirmedReleaseDate,
-          offender.nomisRecord.actualParoleDate
+          offender.prisoner.paroleEligibilityDate,
+          offender.prisoner.conditionalReleaseDate,
+          offender.prisoner.confirmedReleaseDate,
+          offender.prisoner.actualParoleDate
         )
       )
 
     if (!eligibleOffenders.length) return eligibleOffenders
 
     const hdcStatuses = await this.prisonerService.getHdcStatuses(
-      eligibleOffenders.map(c => c.nomisRecord),
+      eligibleOffenders.map(c => c.prisoner),
       user
     )
 
     return eligibleOffenders.filter(offender => {
-      const hdcRecord = hdcStatuses.find(hdc => hdc.bookingId === offender.nomisRecord.bookingId)
+      const hdcRecord = hdcStatuses.find(hdc => hdc.bookingId === offender.prisoner.bookingId)
       return (
-        !hdcRecord ||
-        hdcRecord.approvalStatus !== 'APPROVED' ||
-        !offender.nomisRecord.homeDetentionCurfewEligibilityDate
+        !hdcRecord || hdcRecord.approvalStatus !== 'APPROVED' || !offender.prisoner.homeDetentionCurfewEligibilityDate
       )
     })
   }
 
-  private buildCreateCaseload = (managedOffenders: Array<ManagedCase>): Array<ManagedCase> => {
-    return managedOffenders
-      .filter(
-        offender =>
-          offender.nomisRecord.status &&
-          (offender.nomisRecord.status.startsWith('ACTIVE') || offender.nomisRecord.status === 'INACTIVE TRN')
-      )
-      .filter(offender =>
-        moment(
-          moment(offender.nomisRecord.confirmedReleaseDate || offender.nomisRecord.conditionalReleaseDate, 'YYYY-MM-DD')
-        ).isSameOrAfter(moment(), 'day')
-      )
-      .filter(offender =>
-        [
-          LicenceStatus.OOS_RECALL,
-          LicenceStatus.OOS_BOTUS,
-          LicenceStatus.NOT_IN_PILOT,
-          LicenceStatus.NOT_STARTED,
-          LicenceStatus.IN_PROGRESS,
-          LicenceStatus.SUBMITTED,
-          LicenceStatus.APPROVED,
-          LicenceStatus.TIMED_OUT,
-        ].some(status => offender.licences.find(l => l.status === status))
-      )
+  private filterExistingLicencesForEligibility = (licences: CaCase[]): CaCase[] => {
+    return licences.filter(l => l.nomisLegalStatus !== 'DEAD')
   }
 
-  private pairDeliusRecordsWithNomis = async (
-    managedOffenders: Array<DeliusRecord>,
-    user: User
-  ): Promise<Array<ManagedCase>> => {
-    const caseloadNomisIds = managedOffenders
-      .filter(offender => offender.otherIds?.nomsNumber)
-      .map(offender => offender.otherIds?.nomsNumber)
+  private async mapCasesToComs(cases: CaCase[]): Promise<CaCase[]> {
+    const splitCases = this.splitCasesByComDetails(cases)
```

Comment:
**suggestion:** I found this a little confusing, but like the idea of it. 

Maybe you call this `cases` and prefix the fields `with`.
e.g:
`cases.withNoComId.map(c => ...`

🤷

Reply:
Good shout!